internal/topology/variables/cluster_variable_validation.go

```diff
@@ -56,6 +59,13 @@ func validateClusterVariables(values []clusterv1.ClusterVariable, definitions []
 		return append(allErrs, field.Invalid(fldPath, "["+strings.Join(valueStrings, ",")+"]", fmt.Sprintf("cluster variables not valid: %s", err)))
 	}
 
+	// Get a map of old ClusterVariable values, assume they are all valid and not duplicate names, etc.
```

Comment:
Yes via custom validation webhook, see https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/topology/variables/clusterclass_variable_validation.go#L55

Reply:
That should have been `+listType=map` and `+listMapKey=name` but, ok... Can come back to fix that.

Nit here I would change the language to assert they aren't unique because of validation, rather than "assume"

Reply:
Good idea! Done.

Reply:
> That should have been +listType=map and +listMapKey=name but, ok... Can come back to fix that.

Not sure if that's that easy. name is not unique, name + definitionFrom is unique, but defininitionFrom is optional. Would definitely like to see this fixed, but no idea how :)